src/module/actor/modifiers.ts

```diff
@@ -409,8 +411,32 @@ class StatisticModifier {
     notes?: RollNotePF2e[];
 
     adjustments?: DegreeOfSuccessAdjustment[];
-    /** Allow decorating this object with any needed extra fields. <-- ಠ_ಠ */
-    [key: string]: any;
+
+    /* Extra fields for different Actor classes */
+    value = 0;
+    rank: ZeroToFour = 0;
+
+    // Creature
+    ability?: string;
+    total?: number;
+    type?: string;
+    visible?: boolean;
+
+    // Character
+    expanded?: string | Embedded<ItemPF2e>;
+    itemID?: string;
+    lore?: boolean;
+    rankName?: string;
+    shortform?: string;
+
+    // NPC
+    adjustedHigher?: boolean;
+    adjustedLower?: boolean;
+
+    isLore?: boolean;
+
+    // Hazard
+    base?: number;
```

Comment:
I think most of these can be defined in the extended interface of `StatisticModifier`. Many already are.

Reply:
There is an interface for `RawModifier` that is implemented by `ModifierPF2e` but I can't find an interface for `StatisticModifier`.

Reply:
The properties you added are on the `StatisticModifier` class.